documentation/src/docs/retention.md

```diff
@@ -0,0 +1,155 @@
+        ![CBR truncate Overlap Max Limit](img/Figure7.png)
+
+**Note**: Stream truncation happens asynchronously and eventually based on the truncation Stream-Cuts published by all Subscribers and min/max limits set in the Retention Policy.
+
+#### Enabling Consumption Based Retention on a Pravega Stream
+
+To enable consumption-based retention the following must hold:
+   - The Stream must have a retention policy configured.
+   - One or more Reader Groups reading from the Stream should be `subscriber` Reader Groups.
+   - The Controller should periodically receive acknowledgement Stream cuts from the subscriber Reader Groups, indicating their consumption boundary in the Stream.
+
+#### Creating a Subscriber Reader Group
+
+   A `Retention Type` field in the Reader Group configuration indicates that a Reader Group is either a subscriber or non-subscriber. 
+
+   A `Retention Type` field can have the following values:
+
+   - Auto Publish At Last Checkpoint - A Subscriber Reader Group would automatically publish the Stream Cut corresponding to the last checkpoint to Controller.
+
+   - Manual Publish Stream Cut – A Subscriber Reader Group would not automatically publish Stream Cuts to Controller but instead the user application would need to explicitly invoke `readerGroup.updateTruncationStreamCut()` API and provide a Stream Cut to Controller.
+     
+   - None – A non-subscriber Reader Group. This Reader Group is not expected to publish Stream Cuts to Controller as its read positions do not impact data retention on the Stream.
+     
+   A Reader Group can be converted from a subscriber to non-subscriber or vice versa by changing the corresponding value of the retention policy in the reader group configuration.
+
+#### Acknowledging Consumed Positions Using Checkpoints
+
+   There are two ways for an application to acknowledge its consumed position: automatically via checkpoints or explicitly by generating a Stream Cut and publishing it to the Controller.
+  - Automatically Publish Acknowledgement Stream Cuts:
+    Reader groups checkpoint either automatically or via explicit API calls. A checkpoint consists of coordinating the position across all readers in the group and all segments they are reading from, at checkpoint time. Once a reader learns of an ongoing checkpoint (internally via shared reader group state), it emits a checkpoint event. The checkpoint event for a given reader in the group separates the data events before the checkpoint and events after the checkpoint. Reading a data event after the checkpoint event indicates that the corresponding reader has read and consumed all events before the checkpoint. Once all readers read beyond their corresponding checkpoint events, the Stream Cut corresponding to the position of the checkpoint is published to the Controller as an acknowledgement for the consumed position of this Reader Group.
```

Comment:
Checkpoint is another Pravega concept, so it should be in capitals as well in the document (`checkpoint` -> `Checkpoint`).